src/main/java/com/uid2/admin/vertx/service/EnclaveIdService.java

```diff
@@ -95,35 +95,56 @@ private void handleEnclaveAdd(RoutingContext rc) {
             // refresh manually
             enclaveIdProvider.loadContent(enclaveIdProvider.getMetadata());
 
-            final String name = rc.queryParam("name").get(0);
-            Optional<EnclaveIdentifier> existingEnclaveId = this.enclaveIdProvider.getAll()
-                    .stream().filter(e -> e.getName().equals(name))
-                    .findFirst();
-            if (existingEnclaveId.isPresent()) {
-                ResponseUtil.error(rc, 400, "enclave existed");
+            final String name = rc.queryParam("name").stream().findFirst().orElse(null);
+            if (name == null || name == "") {
+                ResponseUtil.error(rc, 400, "no name specified");
                 return;
             }
 
-            String protocol = RequestUtil.validateOperatorProtocol(rc.queryParam("protocol").get(0));
-            if (protocol == null) {
+            final String protocolString = rc.queryParam("protocol").stream().findFirst().orElse(null);
+            if (protocolString == null || protocolString == "") {
                 ResponseUtil.error(rc, 400, "no protocol specified");
                 return;
             }
 
-            final String enclaveId = rc.queryParam("enclave_id").get(0);
-            if (enclaveId == null) {
+            String protocol = RequestUtil.validateOperatorProtocol(protocolString);
+            if (protocol == null) {
+                ResponseUtil.error(rc, 400, "invalid protocol specified");
+                return;
+            }
+
+            final String enclaveId = rc.queryParam("enclave_id").stream().findFirst().orElse(null);
+            if (enclaveId == null || enclaveId == "") {
                 ResponseUtil.error(rc, 400, "enclave_id not specified");
                 return;
             }
 
-            List<EnclaveIdentifier> enclaveIds = this.enclaveIdProvider.getAll()
-                    .stream().sorted((a, b) -> (int) (a.getCreated() - b.getCreated()))
-                    .collect(Collectors.toList());
+            long created = Instant.now().getEpochSecond();
```

Comment:
nit: Can we use https://github.com/IABTechLab/uid2-admin/blob/main/src/main/java/com/uid2/admin/store/Clock.java instead of `Instant.now()` just to be consistent across the repo. We use the customised Clock so that we are able to mock those values in the tests.

Reply:
Done